cargo-shuttle/src/provisioner_server.rs

```diff
@@ -412,3 +619,107 @@ fn db_type_to_config(db_type: Type) -> EngineConfig {
         },
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use aws_sdk_dynamodb::error::SdkError;
+    use aws_sdk_dynamodb::operation::create_table::CreateTableError;
+    use aws_sdk_dynamodb::operation::create_table::CreateTableOutput;
+    use aws_sdk_dynamodb::operation::scan::ScanError;
+    use aws_sdk_dynamodb::operation::scan::ScanOutput;
+    use aws_sdk_dynamodb::types::AttributeDefinition;
+    use aws_sdk_dynamodb::types::KeySchemaElement;
+    use aws_sdk_dynamodb::types::KeyType;
+    use aws_sdk_dynamodb::types::ProvisionedThroughput;
+    use aws_sdk_dynamodb::types::ScalarAttributeType;
+
+    use crate::provisioner_server::LocalProvisioner;
+
+    async fn create_table(
+        dynamodb_client: &aws_sdk_dynamodb::Client,
+        table_name: &str,
+        attribute_name: &str,
+    ) -> Result<CreateTableOutput, SdkError<CreateTableError>> {
+        let attribute_definition = AttributeDefinition::builder()
+            .attribute_name(attribute_name)
+            .attribute_type(ScalarAttributeType::S)
+            .build();
+
+        let key_schema = KeySchemaElement::builder()
+            .attribute_name(attribute_name)
+            .key_type(KeyType::Hash)
+            .build();
+
+        let provisioned_throughput = ProvisionedThroughput::builder()
+            .read_capacity_units(10)
+            .write_capacity_units(5)
+            .build();
+
+        dynamodb_client
+            .create_table()
+            .table_name(table_name)
+            .key_schema(key_schema)
+            .attribute_definitions(attribute_definition)
+            .provisioned_throughput(provisioned_throughput)
+            .send()
+            .await
+    }
+
+    async fn select_from_table(
+        dynamodb_client: &aws_sdk_dynamodb::Client,
+        table_name: &str,
+    ) -> Result<ScanOutput, SdkError<ScanError>> {
+        dynamodb_client.scan().table_name(table_name).send().await
+    }
+
+    #[tokio::test]
+    async fn test_create_and_delete_dynamodb() {
+        let provisioner = LocalProvisioner::new().unwrap();
+
+        let project_name = "test_create_and_delete_dynamodb".to_string();
+
+        let info = provisioner
+            .get_dynamodb_connection_info(&project_name)
+            .await
+            .unwrap();
+
+        std::env::set_var("AWS_ACCESS_KEY_ID", info.aws_access_key_id);
+        std::env::set_var("AWS_SECRET_ACCESS_KEY", info.aws_secret_access_key);
+        std::env::set_var("AWS_REGION", info.aws_default_region);
+
+        let aws_config = aws_config::from_env()
+            .endpoint_url(info.endpoint.unwrap())
+            .load()
+            .await;
```

Comment:
I suspect the environment variables might not be needed. Using `aws_config::ConfigLoader::default().enpoint_url(...)` might avoid confusion here.

Reply:
> I suspect the environment variables might not be needed.

You'll fail validation by `aws_config` without them.

The closest AWS documentation I can quickly find on this is

> The YAML scripts require that you specify an AWS access key and an AWS secret key, but they are not required to be valid AWS keys for you to access DynamoDB local.

https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocal.DownloadingAndRunning.html

Reply:
@paulotten I did this like yesterday, found out you don't need them. Had to add a bunch of deps tho. https://github.com/jonaro00/wallace-minion/commit/257f38e4aef873374a4a6acabce8cfcd3c55f09f#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcR45